Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/BuildTarget/Common/TestImpactBuildGraph.h

```diff
@@ -160,6 +211,101 @@ namespace TestImpact
 
         return *vertex;
     }
+
+    template<typename ProductionTarget, typename TestTarget>
+    void BuildGraph<ProductionTarget, TestTarget>::WalkTargetBuildGraphSet(
+        TargetBuildGraphPointer graph,
+        TargetBuildGraphSetPointer set,
+        const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+        const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const
+    {
+        AZStd::queue<const BuildGraphVertex<ProductionTarget, TestTarget>*> vertexQueue;
+        AZStd::unordered_set<const BuildGraphVertex<ProductionTarget, TestTarget>*> visitedVertices;
+        AZStd::unordered_map<const BuildGraphVertex<ProductionTarget, TestTarget>*, size_t> vertexDistances;
+
+        // Skip visiting the root vertex and start visiting its children instead
+        const auto& parentVertex = GetVertexOrThrow(buildTarget);
+        visitedVertices.insert(&parentVertex);
+        for (const auto* child : parentVertex.*graph.*set)
+        {
+            vertexQueue.push(child);
+            vertexDistances[child] = 1;
+        }
+
+        while (!vertexQueue.empty())
+        {
+            const auto* vertex = vertexQueue.front();
+            vertexQueue.pop();
+
+            if (const auto result = visitor(*vertex, vertexDistances[vertex]);
+                result == BuildGraphVertexVisitResult::AbortGraphTraversal)
+            {
+                return;
+            }
+            else if (result == BuildGraphVertexVisitResult::AbortBranchTraversal)
+            {
+                continue;
+            }
+
+            for (const auto* child : vertex->*graph.*set)
```

Comment:
repeat `AZStd::invoke` can be used here to make the syntax of accessing the data member a bit more familiar

```suggestion
            for (const auto* child : AZStd::invoke(set, AZStd::invoke(graph, *vertex)))
```